server/api.go

```diff
@@ -495,6 +505,116 @@ func (p *Plugin) getYourPrs(w http.ResponseWriter, r *http.Request) {
 	w.Write(resp)
 }
 
+func (p *Plugin) getPrsDetails(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := context.Background()
+
+	var githubClient *github.Client
+
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
+		return
+	} else {
+		githubClient = p.githubConnect(*info.Token)
+	}
+
+	var prList []*PRDetails
+	json.NewDecoder(r.Body).Decode(&prList)
+
+	var prDetails []*PRDetails
+
+	ch := make(chan *PRDetails)
+	for _, pr := range prList {
+		go fetchPRDetails(ctx, githubClient, pr.URL, pr.Number, ch)
+	}
+
+	for range prList {
+		pr := <-ch
+		prDetails = append(prDetails, pr)
+	}
+
+	resp, _ := json.Marshal(prDetails)
+	w.Write(resp)
+}
+
+func fetchPRDetails(ctx context.Context, client *github.Client, prURL string, prNumber int, ch chan *PRDetails) {
+	status := ""
+	requestedReviewers := []*string{}
+	var reviewsList []*github.PullRequestReview = nil
+
+	repoOwner, repoName := getRepoOwnerAndNameFromURL(prURL)
+
+	chReviews := make(chan []*github.PullRequestReview)
+
+	go fetchReviews(ctx, client, repoOwner, repoName, prNumber, chReviews)
+
+	prInfo, _, err := client.PullRequests.Get(ctx, repoOwner, repoName, prNumber)
+
+	if err != nil {
+		mlog.Error(err.Error())
+		ch <- &PRDetails{
+			URL:                prURL,
+			Number:             prNumber,
+			Status:             status,
+			RequestedReviewers: requestedReviewers,
+			Reviews:            reviewsList,
+		}
+		return
+	}
+
+	for _, v := range prInfo.RequestedReviewers {
+		requestedReviewers = append(requestedReviewers, v.Login)
+	}
+
+	statuses, _, err := client.Repositories.GetCombinedStatus(ctx, repoOwner, repoName, prInfo.GetHead().GetSHA(), nil)
```

Comment:
As an aside, I agree with your concerns, @larkox, re: the number of extra fetches here. I wonder if GitHub's GraphQL endpoints might enable fewer, more performant queries.